Optimisation: simplify some of the LCD fullscreen code

gudnimg opened the pull request:
Removed only lines of code that shouldn't have any effect on the UI for the user. Mostly just redundant code.

Change in memory:
Flash: -62 bytes
SRAM: 0 bytes

gudnimg committed:
Don't handle whitespace in a special way

Instead of reading and checking if a character is a whitespace
just print it out onto the LCD. This makes the code a bit less
complex. Whether there is a whitespace character, we always end
up reading from program space, so I doubt this has much performance
impact.

Change in memory:
Flash: -38 bytes
SRAM: 0 bytes

gudnimg committed:
delay_keep_alive calls manage_heater and manage_inactivity

No need to call them again, the for-loop is very quick

Change in memory:
Flash: -10 bytes
SRAM: 0 bytes

gudnimg committed:
Remove duplicate code

msg_next is NULL, then the end of the message has been reached.

Change in memory:
Flash: -24 bytes
SRAM: 0 bytes

leptun commented on Firmware/ultralcd.cpp:
```diff
@@ -2992,23 +2992,15 @@ static const char* lcd_display_message_fullscreen_nonBlocking_P(const char *msg)
-    uint8_t row = 0;
     bool multi_screen = false;
-    for (; row < 4; ++ row) {
-        while (pgm_is_whitespace(msg))
```
I'm not sure it's ok to remove this. What if the word on the previous line finishes with the last character on the last column of the screen? Wouldn't that cause the next line to have a useless space at the beginning?

gudnimg replied:
You're correct. See images below:

`lcd_display_message_fullscreen_P(_i("Averyveryverlongword Press the knob to pr eheat nozzle and continue."));`

MK3 branch test:
![IMG_2731](https://user-images.githubusercontent.com/8218499/206897317-bee493f9-2065-4573-96ab-df1ef0f6f768.jpg)


PR branch test:
![IMG_2729](https://user-images.githubusercontent.com/8218499/206897168-ccb6ecfc-96a2-4708-8986-c49b5ea01316.jpg)

gudnimg replied:
Fixed in aca9e79

gudnimg committed:
Remove whitespace at start of new row

gudnimg committed:
Remove initial cursor setting

The cursor position is controlled by
the for-loop

Change in memory:
Flash: -8 bytes
SRAM: 0 bytes

DRracer commented on Firmware/ultralcd.cpp:
```diff
@@ -3198,9 +3192,6 @@ uint8_t lcd_show_multiscreen_message_with_choices_and_wait_P(
                 return LCD_BUTTON_TIMEOUT;
             }
-            manage_heater();
-            manage_inactivity(true);
```
Why was this removed? Is lcd_show_multiscreen_message_with_choices_and_wait_P always accompanied with some other ways of managing heaters and inactivity?

gudnimg replied:
There is a call to `delay_keep_alive(50)` above. It calls these functions. There is no need to do it twice, that just increases code size.

DRracer replied:
:+1:
